src/rc/checkpath.c

```diff
@@ -69,6 +70,42 @@ const char * const longopts_help[] = {
 };
 const char *usagestring = NULL;
 
+/*
+ * Work around limitation in Linux when calling fchmod() on a file
+ * descriptor opened with O_PATH.
+ */
+static int fchmod_opath(int fd, mode_t mode)
+{
+#ifdef O_PATH
+	char path[50];
```

Comment:
It appears the number of FDs is limited to unsigned long in the kernel, so the max the path could be is 24 chars (then 25 with the NULL). so its fine as is yeah

Reply:
It looks like the posix version of this, which Linux doesn't implement, is O_SEARCH, but I do not have a way to test [1] [2].

Instead of using a hard-coded array of characters for the paths, I would include helpers.h and use xasprintf() to create them.

[1] http://pubs.opengroup.org/onlinepubs/9699919799/functions/open.html
[2] https://stackoverflow.com/questions/44159325/why-is-there-no-o-search-flag-under-linux-for-open-function

Reply:
We know the size of the string cannot exceed 26 bytes on a system with 32-bit ints, or 35 bytes if ints are 64-bits. 50 bytes was just a ballpark overestimate.

A fixed-size buffer allocated on the stack is more efficient than calling malloc and free.

Reply:
sounds reasonable.  can you add a short comment above the static buffer explaining ?